amethyst_utils/src/ortho_camera.rs

```diff
@@ -0,0 +1,103 @@
+pub enum CameraNormalizeMode {
+    /// Using an aspect ratio of 1:1, tries to ajust the matrix values of the camera so
+    /// that the direction opposite to the stretch_direction is always [0,1].
+    /// Scene space can be lost on the specified stretch_direction.
+    Lossy {stretch_direction: Axis2},
+    
+    /// Scales the render dynamically to ensure no space is lost in the [0,1] range on any axis.
+    Shrink,
+}
+
+impl CameraNormalizeMode {
+    /// Get the camera matrix offsets according to the specified options.
+    pub fn camera_offsets(&self, aspect_ratio: f32) -> (f32,f32,f32,f32) {
+        match self {
+            &CameraNormalizeMode::Lossy {ref stretch_direction} => {
+                match stretch_direction {
+                    Axis2::X => {
+                        CameraNormalizeMode::lossy_x(aspect_ratio)
+                    },
+                    Axis2::Y => {
+                        CameraNormalizeMode::lossy_y(aspect_ratio)
+                    },
+                }
+            },
+            &CameraNormalizeMode::Shrink => {
+                if aspect_ratio > 1.0 {
+                    CameraNormalizeMode::lossy_x(aspect_ratio)
+                } else if aspect_ratio < 1.0 {
+                    CameraNormalizeMode::lossy_y(aspect_ratio)
+                } else {
+                    (0.0,1.0,0.0,1.0)
+                }
+            },
+        }
+    }
+    
+    fn lossy_x(aspect_ratio: f32) -> (f32,f32,f32,f32) {
+        let offset = (aspect_ratio - 1.0) / 2.0;
+        (-offset, 1.0 + offset, 0.0, 1.0)
+    }
+
+    fn lossy_y(aspect_ratio: f32) -> (f32,f32,f32,f32) {
+        let offset = (1.0 / aspect_ratio - 1.0) / 2.0;
+        (0.0, 1.0, -offset, 1.0 + offset)
+    }
+}
+
+impl Default for CameraNormalizeMode {
+    fn default() -> Self {
+        CameraNormalizeMode::Shrink
+    }
+}
+
+/// System that automatically changes the camera matrix according to the settings in the `NormalOrthoCamera` attached to the camera entity.
+#[derive(Default)]
+pub struct NormalOrthoCameraSystem {
+    aspect_ratio_cache: f32,
+}
+
+impl<'a> System<'a> for NormalOrthoCameraSystem {
+    type SystemData = (ReadExpect<'a, ScreenDimensions>, WriteStorage<'a, Camera>, ReadStorage<'a, NormalOrthoCamera>);
+    fn run(&mut self, (dimensions, mut cameras, ortho_cameras): Self::SystemData) {
+        let aspect = dimensions.aspect_ratio();
+        if aspect != self.aspect_ratio_cache {
+            self.aspect_ratio_cache = aspect;
+
+            for (mut camera, ortho_camera) in (&mut cameras, &ortho_cameras).join() {
+                let offsets = ortho_camera.camera_offsets(aspect);
+                camera.proj = Ortho {
+                    left: offsets.0,
+                    right: offsets.1,
+                    bottom: offsets.2,
+                    top: offsets.3,
+                    near: 0.1,
+                    far: 1000.0,
```

Comment:
I think the current default is 2000.0, any specific reason to have it be 1000.0 here?

Reply:
I was afraid it might cause issues with float precision in the matrix because of small values. Scaling of sprites are usually 0.005x or similar.

Reply:
I don't think float precision would be an issue here at only a scale of 2x. You'd better follow the engine convention I think.

Reply:
not that dividing by 2 helps :joy:

Reply:
Yeah right?